src/cli/subcommands/car_cmd.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::path::PathBuf;
+
+use cid::Cid;
+use clap::Subcommand;
+use futures::{AsyncRead, Stream, StreamExt, TryStreamExt};
+use fvm_ipld_car::CarHeader;
+use fvm_ipld_car::CarReader;
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+use crate::ipld::CidHashSet;
+
+type BlockPair = (Cid, Vec<u8>);
+
+#[derive(Debug, Subcommand)]
+pub enum CarCommands {
+    Concat {
+        /// A list of `.car` file paths
+        car_files: Vec<PathBuf>,
+        /// The output `.car` file path
+        #[arg(short, long)]
+        output: PathBuf,
+    },
+}
+
+impl CarCommands {
+    pub async fn run(self) -> anyhow::Result<()> {
+        match self {
+            Self::Concat { car_files, output } => {
+                let readers: Vec<_> = futures::stream::iter(car_files)
+                    .then(tokio::fs::File::open)
+                    .map_ok(tokio::io::BufReader::new)
+                    .map_ok(tokio::io::BufReader::compat)
+                    .map_err(fvm_ipld_car::Error::from)
+                    .and_then(CarReader::new)
+                    .try_collect()
+                    .await?;
+
+                let roots = {
+                    let mut roots = vec![];
+                    let mut seen = CidHashSet::default();
+                    for reader in &readers {
+                        for &root in &reader.header.roots {
+                            if seen.insert(root) {
+                                roots.push(root);
+                            }
+                        }
+                    }
+                    roots
+                };
```

Comment:
```suggestion
                let all_roots = readers
                    .iter()
                    .flat_map(|it| it.header.roots.iter())
                    .unique()
                    .cloned()
                    .collect::<Vec<_>>();
```
nonblocking: manual implementation of `unique`.